pkg/reconciler/pipelinerun/resources/validate_params.go

```diff
@@ -84,3 +85,61 @@ func ValidateObjectParamRequiredKeys(pipelineParameters []v1beta1.ParamSpec, pip
 
 	return nil
 }
+
+// ValidateMatrixPipelineParameterTypes validates the type of Parameter for Matrix.Params
+// and Matrix.Include.Params after any replacements are made from Task parameters or results
+// Matrix.Params must be of type array. Matrix.Include.Params must be of type string.
+func ValidateMatrixPipelineParameterTypes(state PipelineRunState) error {
+	for _, rpt := range state {
+		m := rpt.PipelineTask.Matrix
+		if m.HasInclude() {
+			for _, include := range m.Include {
+				for _, param := range include.Params {
+					if param.Value.Type != v1beta1.ParamTypeString {
+						return fmt.Errorf("parameters of type string only are allowed, but got param type %s", string(param.Value.Type))
+					}
+				}
+			}
+		}
+		if m.HasParams() {
+			for _, param := range m.Params {
+				if param.Value.Type != v1beta1.ParamTypeArray {
+					return fmt.Errorf("parameters of type array only are allowed, but got param type %s", string(param.Value.Type))
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ValidateUniqueParams validates there are no duplicate parameters
+// within Matrix.Params or Matrix.Include Params
+func ValidateUniqueParams(state PipelineRunState) error {
```

Comment:
please make this name clear that it's operating on matrix parameters only

```suggestion
func ValidateUniqueParamsInMatrix(state PipelineRunState) error {
```